src/persist-client/src/internal/state.rs

```diff
@@ -3662,7 +3662,7 @@ pub(crate) mod tests {
 
         // When a writer is present, non-writes don't gc.
         let writer_id = WriterId::new();
-        state.collections.compare_and_append(
+        let _ = state.collections.compare_and_append(
```

Comment:
@MaterializeInc/persist This is necessary because `ControlFlow` now has a `#[must_use]` annotation. Could you please check if ignoring the result is actually fine here?

Reply:
I think as long as the test is passing this is fine, but feel free to assert on the result as well... IIUC this should be deterministic.